src/plugin.ts

```diff
@@ -7,51 +7,63 @@ import { serializeError, type ErrorObject } from 'serialize-error'
 
 import { adjustCanvas, parseImage } from './utils/image'
 import { logger } from './utils/logger'
-import { type DiffOption } from './command'
 
-export type UpdateSnapshotOptions = {
-  screenshotName: string
-  specName: string
-  screenshotAbsolutePath: string
-  baseDirectory?: string
-}
+export type DiffOption = 'always' | 'fail' | 'never'
 
-export type CompareSnapshotsOptions = {
+export type VisualRegressionOptions = {
+  type: string
   screenshotName: string
   errorThreshold: number
   specName: string
   screenshotAbsolutePath: string
   baseDirectory?: string
   diffDirectory?: string
   generateDiff?: DiffOption
+  failSilently: boolean
 }
 
-export type CompareSnapshotResult = {
+export type UpdateSnapshotOptions = Pick<
+  VisualRegressionOptions,
+  'screenshotName' | 'specName' | 'screenshotAbsolutePath' | 'baseDirectory'
+>
+export type CompareSnapshotOptions = Pick<
+  VisualRegressionOptions,
+  | 'screenshotName'
+  | 'specName'
+  | 'screenshotAbsolutePath'
+  | 'baseDirectory'
+  | 'diffDirectory'
+  | 'errorThreshold'
+  | 'generateDiff'
+>
```

Comment:
Nice !! I see that you are using now the [utility types](https://www.typescriptlang.org/docs/handbook/utility-types.html), but maybe in this case (since you are picking more keys than omitting) you could use this one:

```suggestion
export type CompareSnapshotOptions = Omit<VisualRegressionOptions, 'type' | 'failSilently'>
```

😅😅😅. .. I know some times TS is hard to pick the right utility